src/docker.cc

```diff
@@ -32,16 +34,91 @@ namespace google {
 namespace {
 
 #if 0
-constexpr const char docker_endpoint_host[] = "unix://%2Fvar%2Frun%2Fdocker.sock/";
-constexpr const char docker_api_version[] = "1.23";
+constexpr const char kDockerEndpointHost[] = "unix://%2Fvar%2Frun%2Fdocker.sock/";
+constexpr const char kDockerApiVersion[] = "1.23";
 #endif
-constexpr const char docker_endpoint_path[] = "/containers";
+constexpr const char kDockerEndpointPath[] = "/containers";
+constexpr const char kDockerContainerResourcePrefix[] = "container";
 
 }
 
 DockerReader::DockerReader(const MetadataAgentConfiguration& config)
     : config_(config), environment_(config) {}
 
+MetadataUpdater::ResourceMetadata DockerReader::GetContainerMetadata(
+    const json::Object* container, Timestamp collected_at) const
+    throw(json::Exception) {
+  const std::string zone = environment_.InstanceZone();
+
+  const std::string id = container->Get<json::String>("Id");
+  // Inspect the container.
+  try {
+    json::value raw_docker = QueryDocker(
```

Comment:
Being that we're querying for metadata, `raw_metadata`, `raw_container`, or `raw_container_metadata` may be clearer.